lldb/source/Plugins/ObjectFile/Mach-O/ObjectFileMachO.cpp

```diff
@@ -4297,47 +4296,6 @@ void ObjectFileMachO::ParseSymtab(Symtab &symtab) {
 
       if (symbol_section) {
         const addr_t section_file_addr = symbol_section->GetFileAddress();
-        if (symbol_byte_size == 0 && function_starts_count > 0) {
```

Comment:
This block of code is duplicated at line 3437 (only used when lldb is running on an iOS etc device), please make the same removal there.  (along with removing `symbol_size` there, too)